test/e2e/apps/statefulset.go

```diff
@@ -387,7 +387,7 @@ var _ = SIGDescribe("StatefulSet", func() {
 				Type: appsv1.RollingUpdateStatefulSetStrategyType,
 				RollingUpdate: func() *appsv1.RollingUpdateStatefulSetStrategy {
 					return &appsv1.RollingUpdateStatefulSetStrategy{
-						Partition: pointer.Int32(2),
+						Partition: pointer.Int32(1),
```

Comment:
This problem is when all the old version of pod is deleted and the ne w pod have not running, StatefulSet say the upgrade is complete, I'm assuming that this partition number of 2 here doesn't do anything special,so just simply change the number of copies here to 1 to reproduce it.

Reply:
this test has not been modified since version 1.7 https://github.com/kubernetes/kubernetes/commit/b898cf81a50d7b8db53373bafa9a1a0377c0e394 , that means this change is modifying previous behavior, that is something to worry about.

I can not see a new test demonstrating the issue reported

Reply:
Updated, add the new test for it.